### ISSUE TITLE
"highcharts-angular" causes full "highcharts" library to be included in production bundle

### ISSUE THREAD

**Issue by hakimio** [highcharts/highcharts-angular#278]
### Describe the bug
I've created a custom Highcharts bundle like described in [official Highcharts documentation](https://www.highcharts.com/docs/getting-started/how-to-create-custom-highcharts-files#create-a-custom-master-file). Unfortunately `highcharts-angular` causes full `highcharts` library to be always included in production Angular build.

### Expected behavior
`highcharts-angular` should not force full `highcharts` library inclusion.

#### Steps required to recreate the problem in the demo:
1. Create custom bundle like the following:
`highcharts-custom.ts`
```ts
import Highcharts from 'highcharts/es-modules/Core/Globals';
import Utilities from 'highcharts/es-modules/Core/Utilities.js';
import 'highcharts/es-modules/Core/Renderer/HTML/HTMLElement.js';
import 'highcharts/es-modules/Core/Renderer/HTML/HTMLRenderer.js';
import 'highcharts/es-modules/Core/Axis/PlotLineOrBand.js';
import 'highcharts/es-modules/Series/Line/LineSeries.js';
import 'highcharts/es-modules/Series/Column/ColumnSeries.js';
import 'highcharts/es-modules/Series/Scatter/ScatterSeries.js';
import 'highcharts/es-modules/Series/Pie/PieSeries.js';
import 'highcharts/es-modules/Core/Series/DataLabels.js';
import 'highcharts/es-modules/Core/Responsive.js';

// Utilities
Highcharts.addEvent = Utilities.addEvent;
Highcharts.arrayMax = Utilities.arrayMax;
Highcharts.arrayMin = Utilities.arrayMin;
Highcharts.attr = Utilities.attr;
Highcharts.clearTimeout = Utilities.clearTimeout;
Highcharts.correctFloat = Utilities.correctFloat;
Highcharts.createElement = Utilities.createElement;
Highcharts.css = Utilities.css;
Highcharts.defined = Utilities.defined;
Highcharts.destroyObjectProperties = Utilities.destroyObjectProperties;
Highcharts.discardElement = Utilities.discardElement;
Highcharts.erase = Utilities.erase;
Highcharts.error = Utilities.error;
Highcharts.extend = Utilities.extend;
Highcharts.extendClass = Utilities.extendClass;
Highcharts.find = Utilities.find;
Highcharts.fireEvent = Utilities.fireEvent;
Highcharts.format = Utilities.format;
Highcharts.getMagnitude = Utilities.getMagnitude;
Highcharts.getStyle = Utilities.getStyle;
Highcharts.inArray = Utilities.inArray;
Highcharts.isArray = Utilities.isArray;
Highcharts.isClass = Utilities.isClass;
Highcharts.isDOMElement = Utilities.isDOMElement;
Highcharts.isFunction = Utilities.isFunction;
Highcharts.isNumber = Utilities.isNumber;
Highcharts.isObject = Utilities.isObject;
Highcharts.isString = Utilities.isString;
Highcharts.keys = Utilities.keys;
Highcharts.merge = Utilities.merge;
Highcharts.normalizeTickInterval = Utilities.normalizeTickInterval;
Highcharts.numberFormat = Utilities.numberFormat;
Highcharts.objectEach = Utilities.objectEach;
Highcharts.offset = Utilities.offset;
Highcharts.pad = Utilities.pad;
Highcharts.pick = Utilities.pick;
Highcharts.pInt = Utilities.pInt;
Highcharts.relativeLength = Utilities.relativeLength;
Highcharts.removeEvent = Utilities.removeEvent;
Highcharts.splat = Utilities.splat;
Highcharts.stableSort = Utilities.stableSort;
Highcharts.syncTimeout = Utilities.syncTimeout;
Highcharts.timeUnits = Utilities.timeUnits;
Highcharts.uniqueKey = Utilities.uniqueKey;
Highcharts.useSerialIds = Utilities.useSerialIds;
Highcharts.wrap = Utilities.wrap;

export default Highcharts;
```
2. Use it in your application:
```ts
import Highcharts from '../../utils/highcharts-custom';

@Component({
    selector: 'app-bar-chart',
    templateUrl: './bar-chart.component.html',
    styleUrls: ['./bar-chart.component.css']
})
export class BarChartComponent {

    Highcharts = Highcharts;
    // ...
}
```
3. Build production bundle: `ng build --prod`
4. See that `highcharts-angular` pulls full `highcharts` library and the bundle size is even higher than before.

### Additional Info
I have tried recreating `highcharts-angular` library/module in my project and once I replaced the `highcharts` import, my production bundle size went down by approximately ~100KB (depending on your custom bundle, most likely it will be much more).
```ts
import Highcharts from '../../utils/highcharts-custom';
// or use TypeScript type-only import:
// import type * as Highcharts from 'highcharts';

@Component({
    selector: 'highcharts-chart',
    template: ''
})
export class NgxHighchartsComponent implements OnDestroy, OnChanges {

    @Input() Highcharts: typeof Highcharts;
    @Input() constructorType: string;
    @Input() callbackFunction: Highcharts.ChartCallbackFunction;
    @Input() oneToOne: boolean; // #20
    @Input() runOutsideAngular: boolean; // #75
    @Input() options: Highcharts.Options;
    @Input() update: boolean;

    @Output() updateChange = new EventEmitter<boolean>(true);
    @Output() chartInstance = new EventEmitter<Highcharts.Chart>(); // #26

    private chart: Highcharts.Chart;

    // ...
```

### Suggested workaround

Importing highcharts using `type` only imports introduced in [TypeScript 3.8](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) seems to fix the issue for me:
```ts
import type * as Highcharts from 'highcharts';
```

### Setup used
 - NodeJS version: 12
 - Angular version: 11
 - TypeScript version: 4.1

**Comment by hakimio** [comment]
@KacperMadej @mateuszkornecki what's your opinion on this? Are you willing to sacrifice older Angular version support for this?

**Comment by mateuszkornecki** [comment]
@hakimio Thanks for your contribution!

 It looks really interesting, however before we can make any statements we need to test your solution and check how it is affecting the bundle size and how we could use it with an older versions.

**Comment by hakimio** [comment]
@mateuszkornecki It will only effect the bundle size when using custom Highcharts bundles. The TypeScript "type only" imports are unfortunately not compatible with TypeScript lower than version 3.8 which means Angular 9. If you are not willing to sacrifice compatibility with older TypeScript versions, feel free to close this and we can wait until [Highcharts has proper tree-shaking support](https://github.com/highcharts/highcharts/issues/8667#issuecomment-808066989).

**Comment by mateuszkornecki** [comment]
Thanks for the detailed explanation!

 I am afraid that for now, we cannot drop compatibility for the Angular 9< version. One of the main Highcharts features is compatibility (we are still partially supporting IE6 for Highcharts itself... and Angular 8 is not that old at all).

**Comment by hakimio** [comment]
Fair enough.
If anyone else would like to use `highcharts-angular` with type only Highcharts import, I've created custom package: https://github.com/hakimio/highcharts-angular/packages/700261

To use it, you have to add `.npmrc` file next to your package.json with the following contents:
```
@hakimio:registry=https://npm.pkg.github.com
```

**Comment by PowerKiKi** [comment]
For what is worth, Angular 8 is not supported by the Angular team anymore. The [oldest supported release is Angular 9](https://angular.io/guide/releases#support-policy-and-schedule). So the trick explained here could (and IMHO should) be used and would remains with 100% of officially supported code.

While Highchart still support IE6, it is not relevant in this context, because Angular itself does not support it anymore. And as a matter of fact, Angular will [very soon totally drop support for IE 11](https://github.com/angular/angular/issues/41840).

And of course for users that are still on an unsupported version of Angular (8 and older), they could this stay on the previous version of this lib until they upgrade Angular and this lib together.

**Comment by tomgruszowski** [comment]
Fully agree with @PowerKiKi, this backwards compatibility reasoning seems very conservative. If someone is on a legacy version of Angular, there's a high likelihood they also haven't updated High charts.

**Comment by hakimio** [comment]
Seems to be done in #305

**Comment by karolkolodziej** [comment]
Resolved by #305
